### PR TITLE
udf: replace function

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

replace is an user defined function, which is to replace all old substrings with a new substring in a string, as follow: 
mysql> select replace("http://www.baidu.com:9090", "9090", "");
+------------------------------------------------------+
| replace('http://www.baidu.com:9090', '9090', '') |
+------------------------------------------------------+
| http://www.baidu.com:                                |
+------------------------------------------------------+

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/sql-reference/sql-functions/string-functions/str_replace.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+{
+    "title": "str_replace",
```
[the remark this answers is not in the document]
Comment (xinghuayu007) [reply docs/zh-CN/sql-reference/sql-functions/string-functions/str_replace.md:3]: replace is a keyword has been used by sql syntax analysis.

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
I tested the name `replace` and it works. So I suggest to rename the `str_replace` to `replace`.

You can just patch the following `diff` file to your code.

```
diff --git a/fe/fe-core/src/main/cup/sql_parser.cup b/fe/fe-core/src/main/cup/sql_parser.cup
index cd4ea3cdb..7b5e8d2d2 100644
--- a/fe/fe-core/src/main/cup/sql_parser.cup
+++ b/fe/fe-core/src/main/cup/sql_parser.cup
@@ -4592,6 +4592,8 @@ keyword ::=
     {: RESULT = id; :}
     | KW_REPEATABLE:id
     {: RESULT = id; :}
+    | KW_REPLACE:id
+    {: RESULT = id; :}
     | KW_REPLACE_IF_NOT_NULL:id
     {: RESULT = id; :}
     | KW_REPOSITORY:id
diff --git a/gensrc/script/doris_builtins_functions.py b/gensrc/script/doris_builtins_functions.py
index c966b7d8f..ed26fae15 100755
--- a/gensrc/script/doris_builtins_functions.py
+++ b/gensrc/script/doris_builtins_functions.py
@@ -605,7 +605,7 @@ visible_functions = [
             '15FunctionContextENS2_18FunctionStateScopeE'],
     [['concat'], 'VARCHAR', ['VARCHAR', '...'],
             '_ZN5doris15StringFunctions6concatEPN9doris_udf15FunctionContextEiPKNS1_9StringValE'],
-    [['str_replace'], 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'],
+    [['replace'], 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'],
             '_ZN5doris15StringFunctions11str_replaceEPN9doris_udf15FunctionContextERKNS1_9StringValES6_S6_'],
     [['concat_ws'], 'VARCHAR', ['VARCHAR', 'VARCHAR', '...'],
             '_ZN5doris15StringFunctions9concat_wsEPN9doris_udf'
```

**Comment by xinghuayu007** [comment]
> I tested the name `replace` and it works. So I suggest to rename the `str_replace` to `replace`.
> 
> You can just patch the following `diff` file to your code.
> 
> ```
> diff --git a/fe/fe-core/src/main/cup/sql_parser.cup b/fe/fe-core/src/main/cup/sql_parser.cup
> index cd4ea3cdb..7b5e8d2d2 100644
> --- a/fe/fe-core/src/main/cup/sql_parser.cup
> +++ b/fe/fe-core/src/main/cup/sql_parser.cup
> @@ -4592,6 +4592,8 @@ keyword ::=
>      {: RESULT = id; :}
>      | KW_REPEATABLE:id
>      {: RESULT = id; :}
> +    | KW_REPLACE:id
> +    {: RESULT = id; :}
>      | KW_REPLACE_IF_NOT_NULL:id
>      {: RESULT = id; :}
>      | KW_REPOSITORY:id
> diff --git a/gensrc/script/doris_builtins_functions.py b/gensrc/script/doris_builtins_functions.py
> index c966b7d8f..ed26fae15 100755
> --- a/gensrc/script/doris_builtins_functions.py
> +++ b/gensrc/script/doris_builtins_functions.py
> @@ -605,7 +605,7 @@ visible_functions = [
>              '15FunctionContextENS2_18FunctionStateScopeE'],
>      [['concat'], 'VARCHAR', ['VARCHAR', '...'],
>              '_ZN5doris15StringFunctions6concatEPN9doris_udf15FunctionContextEiPKNS1_9StringValE'],
> -    [['str_replace'], 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'],
> +    [['replace'], 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'],
>              '_ZN5doris15StringFunctions11str_replaceEPN9doris_udf15FunctionContextERKNS1_9StringValES6_S6_'],
>      [['concat_ws'], 'VARCHAR', ['VARCHAR', 'VARCHAR', '...'],
>              '_ZN5doris15StringFunctions9concat_wsEPN9doris_udf'
> ```

Thxs!

**Comment by morningman** [comment]
Hi @xinghuayu007 , please let know if you finish modifying this PR, thank you.

**Comment by xinghuayu007** [comment]
> Hi @xinghuayu007 , please let know if you finish modifying this PR, thank you.

I have fixed PR.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**